### PR TITLE
Add Support for Null Usernames in Web Cmdlet Basic Auth

### LINKED ISSUE #9017: Web Cmdlets don't allow $null username in a PSCredential for basic auth
It would be great if there was a way to pass a PSCredential with a $null username to the web cmdlets. I ran into this with an API that has no username, just a API key that is used for basic auth.

# Steps to reproduce

```powershell
$TestCreds = Get-Credential

if ($TestCreds.UserName) {
    $TestCreds = New-Object System.Management.Automation.PSCredential -ArgumentList (
        [pscustomobject] @{ 
            UserName = $null
            Password = $TestCreds.Password
        }
    )
}

Invoke-RestMethod -Uri 'https://httpbin.org/headers' `
                  -Authentication Basic `
                  -Credential $TestCreds
```

# Expected behavior

```powershell
$TestCreds = Get-Credential

if ($TestCreds.UserName) {
    $TestCreds = New-Object System.Management.Automation.PSCredential -ArgumentList (
        [pscustomobject] @{ 
            UserName = ' '
            Password = $TestCreds.Password
        }
    )
}

Invoke-RestMethod -Uri 'https://httpbin.org/headers' `
                  -Authentication Basic `
                  -Credential $TestCreds
```

There is a workaround where you pass a username with a single space. The space is base64 encoded however. So, in the example below, I passed 'username' and 'password' as the creds. You'll notice that 'IDpwYXNzd29yZA==' was passed in the basic auth header, which decodes to ' :password' (note the leading space).

```none
PowerShell credential request
Enter your credentials.
User: username
Password for user username: ********

headers
-------
@{Authorization=Basic IDpwYXNzd29yZA==; Host=httpbin.org; User-Agent=Mozilla/5.0 (Windows NT 10.0; Microsoft Windows 10.0.17134; en-US) PowerShell/6.1.2}
```
# Actual behavior

```none
Invoke-RestMethod : Object reference not set to an instance of an object.
At C:\Users\bdavis\Documents\Scripts\temp_01.ps1:12 char:1
+ Invoke-RestMethod -Uri 'https://httpbin.org/headers' `
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Invoke-RestMethod], NullReferenceException
+ FullyQualifiedErrorId : System.NullReferenceException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.1.2
PSEdition                      Core
GitCommitId                    6.1.2
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by markekraus**
# PR Summary

It is possible to construct PSCredential objects with null or empty usernames. However, doing so makes it impossible to retrieve the plain-text password from the PSCredential.GetNetworkCredential() API. There are some Web APIs which may require a null or empty username. This PR changes the way the password is parsed from the Credential parameter such that a null or empty username will not impact conversion to plain-text.

## PR Context

Closes #9017

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -804,7 +804,8 @@ private bool IsCustomMethodSet()
 
         private string GetBasicAuthorizationHeader()
         {
-            string unencoded = string.Format("{0}:{1}", Credential.UserName, Credential.GetNetworkCredential().Password);
+            var password = new NetworkCredential(null,Credential.Password).Password;
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]:
```suggestion
            var password = new NetworkCredential(null, Credential.Password).Password;
```

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by iSazonov** [comment]
@SteveL-MSFT CodeFactor helped us with such formatting issues. Is it possible to bring it back?

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@iSazonov seems odd that Codacy wouldn't also catch such an issue

**Review by adityapatwardhan — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@markekraus Thank you for your contribution.

**Comment by iSazonov** [comment]
@SteveL-MSFT seems odd that Codacy wouldn't also catch such an issue

Sorry, I don't understand that you mean.
CodeFactor greatly motivated us. We fixed most of style issues. Now there are mainly documentation issues. Since we are generating XML documentation, we should fix such issues as well. We could create something like "continues documentations" so that the XML documentation transparently flows into Docs repo. Therefore, I never wanted to turn off doc issues in CodeFactor. (Although if we turn this off we will have very few reported issues)
On the other hand, Codacy also sometimes looks very strange. There are sometimes false positives.
Together they (CodeFActor and Codacy) would be good.

**Comment by iSazonov** [comment]
Without CodeFactor, we will begin again to collect formatting errors and fall into discussions of this.  
Each reviewer will make conflicting requests. 
It will be sad this is a great job of correcting the formatting to be destroyed.
We already have some such PRs. I really don't understand why CodeFactor was removed if it motivated us to correct above 60000 (!) style issues and make all PRs style clean.

**Pull request merged**